src/pages/dataTableUtilities/actions/cellActions.js

```diff
@@ -286,12 +287,24 @@ export const enforceReasonChoice = (rowKey, route) => (dispatch, getState) => {
   if (!objectToEdit) return null;
 
   const { difference } = objectToEdit;
+
   // If there's no difference, just remove the reason
   if (!difference) return dispatch(removeReason(rowKey, route));
 
   const { validateReason } = objectToEdit;
+
   if (!validateReason) {
-    return dispatch(openModal(MODAL_KEYS.ENFORCE_STOCKTAKE_REASON, rowKey, route));
+    // If updated counted quantity in stocktake page, set batches with default
+    // adjustment reason.
+    if (route === ROUTES.STOCKTAKE_EDITOR) {
+      const [defaultReason] = UIDatabase.objects(
+        difference > 0 ? 'PositiveAdjustmentReason' : 'NegativeAdjustmentReason'
```

Comment:
I was trying to test this part of the requirement: *allow opening the modal without a reason currently selected (currently you can't open the modal without one selected)*

and I can't because there is always the first item in the list selected 😜 

I would have expected that we'd have a blank here, with the user needing to select a reason. I've been told many times to not default a user input, such as an explanatory reason, as that would skew the values. Better to have nothing than a pre-populated value.

Reply:
I agree - but when this happens because we aren't applying the default reason, the enforcement is going to start again. 

When using reasons the action changes : L205-207 in `StocktakeEditPage` - if we don't use the `withReasons` action, just the regular reason, then the enforcement won't happen. 

The enforcing still happens on a batch too - that's from a convoluted process of getting the actions from `getPageActions` which is called from `usePageReducer`. If we just pass the base page actions in `usePageReducer`, we can delete `getPageActions` completely, as well as the actions for editing a counted quantity with reasons and the `enforceReasonChoice` action! 

But this will mean a small adjustment to `DropDownCell`, to show the icon even when no value is set

Reply:
> the enforcement is going to start again

I know! I was avoiding saying that because I thought it would create trouble again.
That all sounds good though. Sorry @wlthomson this may have blown the scope out a little - but it will be awesome if you can do it!